Assets/Scripts/Gameplay/Weapons/WeaponInitiator.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System;
+using Gameplay.Services.FX;
+using UnityEngine;
+using Zenject;
+
+namespace Gameplay.Weapons
+{
+    public class WeaponInitiator : MonoBehaviour
+    {
+        [SerializeField] private Weapon _weaponPrefab;
+        [SerializeField] private Character _character;
+        
+        private PlayingFX.Factory _fxFactory;
+        
+        public event Action ChangeWeaponUIView;
+
+        
+        [Inject]
+        private void Construct(PlayingFX.Factory fxFactory)
+        {
+            _fxFactory = fxFactory;
+        }
+
+        private void Update()
+        {
+            if (Input.GetKeyDown(KeyCode.H))
```

Comment:
Let's improve inputmap and use it for these actions instead of direct keycodes

Reply:
This is debug script. Current implementation normal. I marked script debug (rename)